kivy/ext/mpl/backend_kivy.py

```diff
@@ -0,0 +1,834 @@
+                 dash_joint=gc.line['joint_style'],
+                 dash_list=gc.line['dash_list'])
+
+    def draw_image(self, gc, x, y, im):
+        x = self.widget.x + x
+        y = self.widget.y + y
+        bbox = gc.get_clip_rectangle()
+        if bbox is not None:
+            l, b, w, h = bbox.bounds
+        else:
+            l = 0
+            b = 0,
+            w = self.widget.width
+            h = self.widget.height
+        h, w = im.get_size_out()
+        rows, cols, image_str = im.as_rgba_str()
+        texture = Texture.create(size=(w, h))
+        texture.blit_buffer(image_str, colorfmt='rgba', bufferfmt='ubyte')
+        with self.widget.canvas:
+            Color(1.0, 1.0, 1.0, 1.0)
+            Rectangle(texture=texture, pos=(x, y), size=(w, h))
+
+    def draw_gouraud_triangle(self, gc, points, colors, transform):
+        RendererBase.draw_gouraud_triangle(self, gc, points, colors, transform)
+        print("Entering draw_gouraud_triangle")
+        assert len(points) == len(colors)
+        assert points.ndim == 3
+        assert points.shape[1] == 3
+        assert points.shape[2] == 2
+        assert colors.ndim == 3
+        assert colors.shape[1] == 3
+        assert colors.shape[2] == 4
+        shape = points.shape
+        points = points.reshape((shape[0] * shape[1], 2))
+        tpoints = trans.transform(points)
+        tpoints = tpoints.reshape(shape)
+        name = self.list_goraud_triangles.append((tpoints, colors))
+        ''' Implement this pseudocode with points and colors:
+        http://www-users.mat.uni.torun.pl/~wrona/3d_tutor/tri_fillers.html '''
+
+    def draw_gouraud_triangles(self, gc, triangles_array, colors_array,
+        transform):
+        print ("Entra en draw_gouraud_triangles")
+#         self.draw_gouraud_triangles(gc, points.reshape((1, 3, 2)),
+#                                     colors.reshape((1, 3, 4)), trans)
+
+    def draw_text(self, gc, x, y, s, prop, angle, ismath=False, mtext=None):
+        x = int(x) + self.widget.x
+        y = int(y) + self.widget.y
+        if x < 0 or y < 0:
+            return
+        if ismath:
+            self.draw_mathtext(gc, x, y, s, prop, angle)
+        else:
+            font = resource_find(prop.get_name() + ".ttf")
+            if font is None:
+                plot_text = CoreLabel(font_size=prop.get_size_in_points())
+            else:
+                plot_text = CoreLabel(font_size=prop.get_size_in_points(),
+                                font_name=prop.get_name())
+            plot_text.text = "{}".format(s)
+            if prop.get_style() == 'italic':
+                plot_text.italic = True
+            if weight_as_number(prop.get_weight()) > 500:
+                plot_text.bold = True
+            plot_text.refresh()
+            with self.widget.canvas:
+                if isinstance(angle, float):
+                    PushMatrix()
+                    Rotate(angle=angle, origin=(x, y))
```

Comment:
matham I get the same result in different backends. The rotation is not from the center for the look of it is right, and the superposition would happen anyway if we don't use CoreLabel.